content/2018-12-07_par-slov-o-datovem-typu-enum.md

```diff
@@ -0,0 +1,192 @@
+Title: Pár slov o datovém typu Enum
+Date: 2018-12-04 20:48:00
+Modified: 2018-12-04 20:48:00
+Author: Karolina Surma
+
+
+# Třída Direction - příklad implementace
+_(dřív zde na blogu pojmenovaná Rotation)_
+
+Na dnešním srazu jsme řešily víc témat, z nichž jeden bylo použití třídy **Direction** datového typu Enum. Protože je to látka nová pro nás všechny na Roboprojektu, zde si podrobněji rozepíšeme, k čemu a jak tuto třídu tohoto typu můžeme použít.
+
+Doposud v projektu třídy byly tři: *Robot*, *Tile* a *State*. Ta poslední obsahuje jak informace o mapě, tak o robotech a jejích koordinátách. Všechny byly napsány podle pravidel nám známých ze [začátečnického kurzu PyLadies](https://naucse.python.cz/course/pyladies/beginners/class/), to znamená, že můžeme (a měly bychom) vytvořit nové objekty dané třídy tak, že ho inicializujeme, např.:
+
+`my_robot = Robot(<direction>, <picture>, <coordinates>)`
+
+Mezi `<>`  výše se nachází příklady argumentů, které musíme dát objektu třídy *Robot*, aby se korektně objevil na mapě (tzn. vykreslil správný obrázek na správném místě mapy se zobáčkem nasměrovaným na sever/ jih/ západ/ východ). 
+
+Třída **Direction** je ale jiná. Zde víme předem, kolik objektů této třídy bude ve hře: 4. Víme, že samotné objekty se nebudou měnit, ale je možné změnit směr robota nebo políčka na nový. Nebudeme inicializovat nové objekty této třídy, ale uložíme si naše směry jako konstanty. Z objektu takové třídy by se měly dát vyčíst různé informace dle tabulky obsažené v [předchozím postu](https://roboprojekt.pyladies.cz/devaty-sraz-upravujeme-navrhujeme). 
+
+O třídě, která přijde jako vhodné řešení tohoto zadání, jsme se již před nějakou dobou bavily. Jedná se o datový typ Enum, importovaný na začátku pythonního souboru:
+
+```python
+from enum import Enum 
+        
+    class Direction(Enum):
+        N = 0
+        E = 90
+        S = 180
+        W = 270
+```
+Pokud si tu třídu naimportujeme do konzoly nebo vyzkoušíme .py soubor, můžeme k jejím objektům přistoupit velice jednoduše:
```

Comment:
..."tuto třídu"...
do konzoly - to je trochu sporné. Nakolik ÚJČ toto slovo ve významu počítačové konzole nezdefinuje, nemá také dané jeho skloňování, zná pouze slovo konzola v jiném významu. V tomto případě je ale podle mě správně "do konzole", alespoň tak se to v počítačovém kontextu podle mě používá.

Reply:
A není to typ "žena"? Konzola - bez konzoly? Nebo bez konzole, jako v typu růže?